### ISSUE TITLE
[CL][Concentrated Liquidity Module]: implement swap keeper methods and connect with the swap-router

### ISSUE THREAD

**Issue by p0mvn** [osmosis-labs/osmosis#3063]
## Background

Swap router modules routes swaps either to `gamm` or `concentrated-liquidity`. Therefore, we need to implement and connect swap methods in the `concentrated-liquidity` module.

## Acceptance Criteria
- swap keeper methods added
- integrated to the `swap-router` module

**Comment by p0mvn** [comment]
blocked by: https://github.com/osmosis-labs/osmosis/issues/3066

**Comment by czarcas7ic** [comment]
@catShaark we are going to keep all concentrated liquidity items in house for now, since we are all co-located to implement this feature. Will let you know if this changes, thanks for picking this up though!

**Comment by catShaark** [comment]
ohhh, thanks for notifying me

**Comment by mattverse** [comment]
@catShaark  thank you for everything!

**Comment by p0mvn** [comment]
Further progress on this blocked by pool id routing. TBD in https://github.com/osmosis-labs/osmosis/issues/3097

Some work towards this has been done in #3128 so I'm going to close is. Will create new issue after #3097 if still applicable
